docs/event-processing/idempotent-writer.md

```diff
@@ -0,0 +1,42 @@
+---
+seo:
+  title: Idempotent Writer
+  description: An Idempotent Writer produces an Event to an Event Streaming Platform exactly once.
+---
+
+# Idempotent Writer
+A writer produces [Events](../event/event.md) that are written into an [Event Stream](../event-stream/event-stream.md), and under stable conditions, each Event is recorded only once.
+However, in the case of an operational failure or a brief network outage, an [Event Source](../event-source/event-source.md) may need to retry writes. This may result in multiple copies of the same Event ending up in the Event Stream, as the first write may have actually succeeded even though the producer did not receive the acknowledgement from the [Event Streaming Platform](../event-stream/event-streaming-platform.md). This type of duplication is a common failure scenario in practice and one of the perils of distributed systems.
+
+## Problem
+How can an [Event Streaming Platform](../event-stream/event-streaming-platform.md) ensure that an Event Source does not write the same Event more than once?
+
+## Solution
+![idempotent-writer](../img/idempotent-writer.svg)
+
+Generally speaking, this can be addressed by native support for idempotent clients.
+This means that a writer may try to produce an Event more than once, but the Event Streaming Platform detects and discards duplicate write attempts for the same Event.
+
+## Implementation
+To make an Apache Kafka® producer idempotent, configure your producer with the following setting:
+
+```
+enable.idempotence=true
+```
+
+The Kafka producer tags each batch of Events that it sends to the Kafka cluster with a sequence number. Brokers in the cluster use this sequence number to enforce deduplication of Events sent from this specific producer. Each batch's sequence number is persisted so that even if the [leader broker](https://www.confluent.io/blog/apache-kafka-intro-how-kafka-works/#replication) fails, the new leader broker will also know if a given batch is a duplicate.
+
+To enable [exactly-once processing guarantees](https://docs.ksqldb.io/en/latest/operate-and-deploy/exactly-once-semantics/) in ksqlDB or Kafka Streams, configure the application with the following setting, which includes enabling idempotence in the embedded producer:
+
+```
+processing.guarantee=exactly_once
```

Comment:
The docs link for ksqlDB above says that it is indeed this setting.

For Kafka Streams, however, the current docs at https://docs.confluent.io/platform/current/streams/developer-guide/config-streams.html#processing-guarantee ask for a different setting:

```
processing.guarantee= exactly_once_v2
```

Reply:
So, there's a bug in our documentation. The correct setting for KSTreams (as of today) is:

```
processing.guarantee=exactly_once_beta
```

The `*_v2` variant will be introduced in AK 3.0 through KIP-732.

Reply:
In this case, I would stick with the v1 version of EOS for now, because it doesn't have the (current) "beta" suffix like the v2 version of EOS.

TL;DR:
```
processing.guarantee=exactly_once
```
(this gives you EOS v1)

Reply:
@miguno to confirm, no change required to this PR?

Reply:
So, no change is needed for now. We can add the `_v2` setting once it's available officially.

Reply:
Yes, no change is required. I already Approved subsequently.